DolevIll/tutorial3/homework3/musicstats.sh

```diff
@@ -0,0 +1,320 @@
+#       and ends with an ARTIST: or COMMENT entry
+#
+# hint: find counter examples that contradict 
+#       each of the above statements (if any).
+
+
+####### CODE START #####################################################
+
+
+# This variable contains the artist names passed as arguments. 
+# Leave untouched. 
+ARTISTNAMES=$@
+
+### E1 (3 points) - Using an if, check that the variable $ARTISTNAMES
+# is not empty and exit with error code 1.
+# If empty, it means there is no artist name as first argument of the 
+# script. 
+# Print an explanatory error message to inform the user of the error.
+# Use the predefined function usage() defined at lines 23-30 of 
+# this file to give the user information how to pass the parameters.
+# DO NOT MODIFY THE usage() FUNCTION BODY.
+
+if [ -z "${ARTISTNAMES}" ] 
+then 
+	 echo "\$ARTISTNAMES is empty" 
+	 usage 
+else 
+       echo "\$ARTISTNAMES is NOT empty" 
+fi
+########################################################################
+### E2 (3 points) Download the STIL.txt db
+# Test if the STIL.txt file exists. If it doesn't exist, write the code 
+# to download it using the wget command. 
+# exit with error if the download fails.
+# if the file already exists, do not download it and continue the
+# execution.
+# Hint: use the -e condition and the predefined variable $?
+
+# E2.1 (2 points) if the STIL.txt file does not exist, download the file
+FILE=STIL.txt
+if [ ! -e "$FILE" ]; then
+    wget -q  https://hvsc.de/download/C64Music/DOCUMENTS/STIL.txt
+   # E2.2 (1 point ) If the wget command fails, exit with error.
+	r='wget -q  https://hvsc.de/download/C64Music/DOCUMENTS/STIL.txt'
+	if [ $r -ne 0 ]; #This gives an error when deleting STIL.txt, why?
+  	then 
+  	echo "Error in executing wget, exiting the script"
+	exit 1
+	fi   
+   # otherwise (if the file exists) do not download anything
+   else 
+   echo "STIL database found, will not download again."
+fi
+
+### E2 END #############################################################
+
+########################################################################
+### E3 (4 points) - Prepare the stats folder 
+# The temporary folder should be called 'stats' and its name is stored in
+# the STATSDIR variable.
+# E3.1 (1 point) Define and instantiate the STATSDIR variable with
+# the string 'stats'
+STATSDIR='stats'
+
+
+# E3.2 (2 points) Use an if to test if the 'stats' folder exists in the
+# folder where the script is being executed.
+# if it does, delete it.
+# Always print out information to the user about what is happening.
+#YOUR_CODE_HERE
+if [ -d $STATSDIR ]; then
+	echo "$STATSDIR exists, deleting previous statistics"
+	read -p "Is this okay?" -n 1 -r
+	echo
+	if [[ $REPLY =~ ^[Yy]$ ]]; then
+	rm -r $STATSDIR
+	fi
+else
+	mkdir $STATSDIR
+fi
+# E3.3 (1 point ) Create a new folder taking the name from the 
+# variable $STATSDIR
+echo "Creating new stats directory $STATSDIR"
+mkdir $STATSDIR	
```

Comment:
Did you test the code at all? the scripts generates an error at this stage because you already created the directory at line 159